src/main/java/com/guysfromusa/carsgame/services/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.guysfromusa.carsgame.services.impl;
+
+import com.guysfromusa.carsgame.entities.CarEntity;
+import com.guysfromusa.carsgame.entities.enums.CarType;
+import com.guysfromusa.carsgame.model.TurnSide;
+import com.guysfromusa.carsgame.repositories.CarRepository;
+import com.guysfromusa.carsgame.services.CarService;
+import org.apache.commons.collections4.IteratorUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.inject.Inject;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+@Service
+@Transactional
+public class CarServiceImpl implements CarService {
+
+    private final CarRepository carRepository;
+
+    @Inject
+    public CarServiceImpl(CarRepository carRepository) {
+        this.carRepository = carRepository;
+    }
+
+    @Override
+    public long deleteCarByName(String name) {
+        return carRepository.deleteByName(name);
+    }
+
+    @Override
+    public Long addCar(CarType carType, String name) {
```

Comment:
lets use small longs always when we are sure that there cant be null, then we will know (without checking) null defence checks are not needed